src/Models/ArduinoDeviceBase.ts

```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
   abstract async configDeviceSettings(): Promise<boolean>;
 
   abstract async load(): Promise<boolean>;
+
+
   abstract async create(): Promise<boolean>;
 
+  async createCore(board: Board|undefined, templateFiles: TemplateFileInfo[]):
+      Promise<boolean> {
+    const deviceFolderPath = this.deviceFolder;
+
+    const scaffoldType = ScaffoldType.Local;
+    if (!await FileUtility.directoryExists(scaffoldType, deviceFolderPath)) {
+      throw new Error('Unable to find the device folder inside the project.');

```

Comment:
Internal error, couldn't find the template folder

Reply:
Fix.